x-pack/plugins/cases/server/common/models/case_with_comments.ts

```diff
@@ -293,9 +313,31 @@ export class CaseCommentModel {
       references = [...references, ...commentStringReferences];
     }
 
+    if (
+      commentReq.type === CommentType.externalReference &&
+      commentReq.externalReferenceStorage.type === ExternalReferenceStorageType.so
+    ) {
+      references = [
+        ...references,
+
+        this.getExternalReferenceSoReferences({
+          id: commentReq.externalReferenceId,
+          type: commentReq.externalReferenceStorage.soType,
+        }),
+      ];
+    }
+
     return references;
   }
 
+  private getExternalReferenceSoReferences({ id, type }: { id: string; type: string }) {
```

Comment:
nit: since this only returns a single object let's rename the function to `getExternalReferenceSoReference` (drop the plural `s`).